src/main/java/frc/robot/poseestimator/limelights/VisionObservationFiltered.java

```diff
@@ -0,0 +1,84 @@
+package frc.robot.poseestimator.limelights;
+
+import frc.robot.constants.Field;
+import frc.robot.poseestimator.observations.VisionObservation;
+import frc.utils.GBSubsystem;
+import org.littletonrobotics.junction.Logger;
+
+import java.util.ArrayList;
+import java.util.List;
+
+public class VisionObservationFiltered extends GBSubsystem {
+
+	private final MultiLimelightsRawData limelightHardware;
+	private final VisionObservationFilteredConfig config;
+
+	public VisionObservationFiltered(VisionObservationFilteredConfig config) {
+		super(config.logPath());
+
+		this.limelightHardware = new MultiLimelightsRawData(config.limelightsNames(), config.hardwareLogPath());
+		this.config = config;
+	}
+
+	public List<VisionObservation> getFilteredVisionObservations() {
+		ArrayList<VisionObservation> estimates = new ArrayList<>();
+
+		for (LimelightRawData limelightRawData : limelightHardware.getAllAvailableLimelightData()) {
+			if (keepLimelightData(limelightRawData)) {
+				double standardTransformDeviation = getDynamicStandardTransformDeviations(limelightRawData);
+				double[] standardDeviations = new double[] {
+					standardTransformDeviation,
+					standardTransformDeviation,
+					VisionConstants.STANDARD_DEVIATION_VISION_ANGLE};
+
+				estimates.add(
+					new VisionObservation(limelightRawData.estimatedPose(), standardDeviations, limelightRawData.timestamp())
+				);
+			}
+		}
+
+		return estimates;
+	}
+
+	private boolean isLimelightOutputInTolerance(LimelightRawData limelightRawData) {
+		// ! THIS SHOULDN'T BE COMMENTED OUT
```

Comment:
> THIS SHOULDN'T BE COMMENTED OUT

~ me

Inject Robot

Reply:
we MUST wait for swerve in order to use SuperStructure for injections

Reply:
Open issue?

Reply:
once I figure out how

Reply:
(issue opend)[https://github.com/GreenBlitz/GB-Robot-Template/issues/369]